### ISSUE TITLE
Editor command line --doctool option stalls on generating new docs

### ISSUE THREAD

**Issue by MylesScholz** [godotengine/godot#94966]
### Tested versions

- Reproducible in: 4.2.2.stable [15073afe3], 4.3.rc [3e0c10d39]

### System information

Godot v4.3.rc (3e0c10d39) - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2060 with Max-Q Design (NVIDIA; 30.0.15.1278) - AMD Ryzen 9 4900HS with Radeon Graphics (16 Threads)

### Issue description

When running the editor with the `--doctool` option from the command line, the program loads the docs and erases the old docs, but it stalls for >1 hour on "Generating new docs...". I assume this is a long-running operation, but the wait time seems excessively long, and there is no indication of progress.

I was looking into updating some class reference documentation by following [this guide](https://docs.godotengine.org/en/latest/contributing/documentation/updating_the_class_reference.html), but it didn't mention how long the doctool is expected to run or if there were any required build options for the doctool to work.

### Steps to reproduce

0. Clone the Godot repository
1. Build with `scons .`
2. Run `.\bin\godot.windows.editor.x86_64.exe --doctool`
3. Wait...

### Minimal reproduction project (MRP)

N/A

**Comment by bruvzg** [comment]
> .\bin\godot.windows.editor.x86_64.exe --doctool

Always use `.console.exe` instead of GUI one when running from command line. Most likely it does not stall, it's just not redrawing command prompt since app is detached from console and shell can't know when GUI app is finished (you can press Enter to force redaw it).

**Comment by MylesScholz** [comment]
I ran `.\bin\godot.windows.editor.x86_64.console.exe --doctool` on the 4.3.rc [3e0c10d] version, and it's still going after over an hour. I tried pressing Enter several times, but that didn't do anything. I'll leave it running for longer to see if it's just slow. I don't know if it's supposed to create the files as it goes, but it hasn't created or modified any.

**Comment by MylesScholz** [comment]
It's been running for 9 hours now with no change.

**Comment by MylesScholz** [comment]
I ran the latest version (4.3.rc [3978628c6]) with gdb, and it seems to get stuck on editor/doc_tools.cpp:1599:
`Ref<FileAccess> f = FileAccess::open(save_file, FileAccess::WRITE, &err);`
The particular class it gets stuck on is inconsistent, and sometimes it gets partially through modifying some files.

**Comment by MylesScholz** [comment]
I don't believe so. It's just on my C drive.

**Comment by MylesScholz** [comment]
Yes, the editor and the console version are running but not doing anything. I will try digging into the FileAccess::open() call with gdb to see where the problem is more specifically.
![TaskManager](https://github.com/user-attachments/assets/e25740df-5eaf-4769-a17b-b1441ad4569d)

**Comment by AThousandShips** [comment]
Check details to see if orphan processes are running

**Comment by MylesScholz** [comment]
I'm not sure how to check for orphan processes, but here is the details page:

![TaskManager1](https://github.com/user-attachments/assets/8ac6be32-35fb-4a10-bf93-d1941d05a7c4)

**Comment by MylesScholz** [comment]
I've isolated the stall to `drivers/windows/file_access_windows.cpp:197`:

`f = _wfsopen((LPCWSTR)(path.utf16().get_data()), mode_string, is_backup_save_enabled() ? _SH_SECURE : _SH_DENYNO);`

Interestingly, sometimes when stepping through the program with gdb, it works all the way through. That might suggest it's a concurrency issue.

**Comment by chsoen** [comment]
I was unable to reproduce the bug on v4.5.dev.custom_build [99f5a3d66] with Windows 11 and Fedora 42
The doctool finishes for me and modifies a bunch of the xml doc files.
```
Godot v4.5.dev (99f5a3d66) - Windows 11 (build 26100) - Multi-window, 1 monitor - Vulkan (Forward+) - dedicated AMD Radeon RX 6750 XT (Advanced Micro Devices, Inc.; 32.0.12033.1030) - 12th Gen Intel(R) Core(TM) i7-12700KF (20 threads)
```
```
Godot v4.5.dev (99f5a3d66) - Fedora Linux 42 (Workstation Edition) on Wayland - X11 display driver, Multi-window, 1 monitor - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1650 (NVK TU117) - AMD Ryzen 5 4600H with Radeon Graphics (12 threads)
```

**Comment by Calinou** [comment]
@MylesScholz Can you (or anyone else) still reproduce this bug in [Godot 4.4.1](https://godotengine.org/article/maintenance-release-godot-4-4-1/) or any later release?

**Comment by Calinou** [comment]
Closing due to lack of response. Please comment if you can still reproduce this bug on the latest Godot version.

**Comment by solodeveloping** [comment]
Maybe the doc should indicate to use console.exe since it's expected, otherwise this issue becomes the documentation.

The same happened to me, but it stall at "Deleting docs cache..." for me.

I don't see an instance of Godot.
